### PR TITLE
Add workaround for libc++ incomplete from_chars().

### PR DESCRIPTION
**Pull request by atgeirr**
Float version not included in current version of libc++(15).

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

**Review by akva2 — APPROVED** [review]

File: opm/models/utils/parametersystem.hh [hunk 1]
```diff
@@ -924,7 +925,19 @@ auto Get(bool errorIfNotRegistered)
...
     else {
+#ifdef _LIBCPP_VERSION // If this macro is defined, clang's libc++ is used
+        // For floating point types, libc++ (the llvm/clang library implementation)
+        // does not yet (as of clang 15) offer an implementation of std::from_chars().
+        if constexpr (std::is_integral_v<ParamType>) {
+            std::from_chars(defVal.data(), defVal.data() + defVal.size(), defaultValue);
+        } else {
+            // Floating point workaround.
+            char* end_ptr = nullptr;
+            defaultValue = std::strtod(defVal.c_str(), &end_ptr);
```
Comment (bska) [inline opm/models/utils/parametersystem.hh:936]: Tiny simplification: It's okay to call `strtod()` as
```C++
defaultValue = std::strtod(defVal.c_str(), nullptr);
```
in this case.

Comment (atgeirr) [reply opm/models/utils/parametersystem.hh]: Done.

**Comment by atgeirr** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge into master.

**Pull request merged**